### DISCUSSION TITLE
yolov5/tutorials/train_custom_data/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#8456] [Docs]
# yolov5/tutorials/train_custom_data/

Learn how to train your data on custom datasets using YOLOv5. Simple and updated guide on collection and organization of images, labelling, model training and deployment.

https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/

**Comment by yassineguenidi** [comment]
can i work with labels personnalized for exemple i work with invoice object detection so labels are invoice_no, .... ?

**Reply by pderrenger** [reply]
Absolutely, you can work with customized labels like `invoice_no` for your invoice object detection project! 🚀 YOLOv8, similar to YOLOv5, is designed to be flexible and can be trained on any dataset as long as it's properly labeled and formatted.

Here's a quick guide to get you started:

1. **Collect and Organize Your Images**: Gather a diverse set of invoice images that represent the variations you expect in the real world.

2. **Label Your Data**: Use a labeling tool to annotate your images with your custom labels (`invoice_no`, etc.). Make sure each object of interest in your images is annotated with a bounding box and the correct label.

3. **Prepare Your Dataset**: Organize your labeled images and annotations in a structure supported by YOLOv8. You'll need a `dataset.yaml` file that specifies paths to your training and validation images, as well as a list of your custom class names.

4. **Train Your Model**: With your dataset ready, you can start training your model. Here's a simple command to kick off training with YOLOv8:

   ```bash
   yolo train data=your_dataset.yaml model=your_model.yaml epochs=100 imgsz=640
   ```

5. **Evaluate and Deploy**: After training, evaluate your model's performance and deploy it for inference on new invoice images.

Remember, the key to a successful object detection model is a well-labeled, diverse dataset. So, take your time with the data preparation step. Happy detecting! 🎉

**Comment by 18Clouds** [comment]
I am more familiar to yolov8 but have to work with and am new to yolov5, but I cannot get a custom v5 model to run using a python command like model = YOLO('./model.pt') . How can I use this model to get bounding box coordinates of objects in the images?

**Reply by yassineguenidi** [reply]
import torch
model = torch.hub.load('.', 'custom', path='/content/yolov5/runs/train/results_1/weights/best.pt', source='local')

img = '/content/test/images/20190925810_jpg.rf.f9e58a3b226d7f3f00366d9531a41e6a.jpg'

results = model(img)
results.show()

u can use sth like that it works for me

**Reply by glenn-jocher** [reply]
Hey there! 😊 It looks like you're trying to use a YOLOv5 model with a syntax that's more aligned with YOLOv8. For YOLOv5, you'll want to use the `torch` library to load your model and then perform inference to get those bounding box coordinates. Here's a quick example to help you out:

```python
import torch
from PIL import Image

# Load your custom model
model = torch.load('./model.pt')
model.eval()  # Set the model to evaluation mode

# Load your image
img = Image.open('path/to/your/image.jpg')

# Perform inference
results = model(img)

# Extract bounding box coordinates
for *xyxy, conf, cls in results.xyxy[0]:
    print(f'Coordinates: {xyxy} | Confidence: {conf} | Class: {cls}')
```

This should get you started with running your YOLOv5 model on your images. If you have any more questions or need further assistance, feel free to ask. Happy coding! 🚀

**Comment by ivanstepanovftw** [comment]
How to enable/disable augmentations when training custom YOLOv5 model from launching `train.py`?

**Reply by ivanstepanovftw** [reply]
I guess it is `[--hyp HYP]` parameter, and examples are in `/yolov5/data/hyps`

**Comment by AsmaMubarakk** [comment]
Is it possible to use the content provided in this repository (specifically, the pretrained model table types mentioned in point #3) for a research paper, ensuring proper citation?

**Reply by pderrenger** [reply]
@AsmaMubarakk hello! Absolutely, you are welcome to use the pretrained model table types mentioned in our repository for your research paper. We appreciate you ensuring proper citation 📚. 

In citing our work, you might refer to the specific GitHub repo URL and indicate it was accessed on [date of access]. If there's a related publication or documentation, citing that as well would be beneficial. For instance:

```
Ultralytics. (Year). YOLOv8 Pretrained Models. [online] Available at: [repository URL] [Accessed Date].
```

If you have any more questions or need further details for your citation, feel free to ask. Happy researching! 🚀

**Comment by zikav1** [comment]
Hi, is it possible to create a YOLOv5 dataset from a CSV-file?

**Reply by yassineguenidi** [reply]
Hello, yolov5 accepts datasets that contain images in one folder and their annotations in another folder so ur CSV file contains for example the path of your image and the annotation of each class,

**Reply by pderrenger** [reply]
Hello! 😊

Yes, you can certainly create a YOLOv5 dataset from a CSV file. Here's a quick guide on how to do it:

1. **Organize your CSV file**: Ensure your CSV file has columns for images paths and associated labels.
2. **Convert CSV to YOLO format**: You'll need a script that reads the CSV and writes each entry's annotations to a separate `.txt` file in YOLO format `(class x_center y_center width height)`, all normalized by image width and height.
3. **Use your data**: Incorporate these labels and images into the YOLOv5 training process by modifying the `dataset.yaml` to point to your training and validation image folders.

Here's a tiny bit of Python pseudocode to get you started:
```python
import pandas as pd
df = pd.read_csv("data.csv")
for index, row in df.iterrows():
    # Process your rows to YOLO format
    with open(f"{row['image_path'][:-4]}.txt", 'w') as file:
        file.write(f"{row['class']} {row['x_center']} {row['y_center']} {row['width']} {row['height']}\n")
```
Just replace the column names with ones corresponding to your CSV.

Let me know if there's anything else you'd like to know!

**Comment by Muhammad-Irfan-Shahzad** [comment]
I am facing this issue "ValueError: malformed node or string: <ast.Name object at 0x0000010EFA2BE550>" when i convert trained yolov5 on custom data and convert into tflite after that add metadata file then I test on image the tflite . I am facing this isssue. I have only one class "ball".

**Reply by pderrenger** [reply]
@Muhammad-Irfan-Shahzad it seems like you're encountering an issue with the conversion of a YOLOv5 model to TensorFlow Lite (TFLite) format and adding metadata. This error typically arises from a problem in the script or the way the model is being handled during conversion or testing.

To assist you better, could you please share the specific code snippet you're using for the conversion to TFLite and the addition of metadata? This will help in pinpointing the exact cause of the error.

Meanwhile, ensure that your environment is set up correctly with all necessary dependencies, and that the model and data paths are correctly specified. Here’s a basic example of how you might convert a YOLOv5 model to TFLite:

```python
import torch
from yolov5 import YOLO

# Load the trained YOLOv5 model
model = YOLO('path/to/your/custom_trained_yolov5_model.pt').autoshape()  # for file/URI/PIL/cv2/np inputs and NMS
model.eval()

# Convert to TorchScript
ts_model = model.model.to_torchscript()

# Save the TorchScript model
ts_model.save('model.torchscript.pt')

# Convert TorchScript model to TensorFlow Lite
converter = torch.jit.load('model.torchscript.pt')
converter = torch.jit.optimize_for_mobile(converter)
converter._save_for_lite_interpreter('model.tflite')
```

Make sure to adjust the paths and model names as per your setup. If you continue to face issues, please provide the error logs and code snippet for more detailed guidance.

**Comment by akhileshkumar1209** [comment]
i am facing the issue in my custom model training i have tried more then 10 times to train the model but in 8 time it is showing corrupt images as error how can i fix it. 
Transferred 342/349 items from yolov5s.pt
AMP: checks failed ❌, disabling Automatic Mixed Precision. See https://github.com/ultralytics/yolov5/issues/7908
optimizer: SGD(lr=0.01) with parameter groups 57 weight(decay=0.0), 60 weight(decay=0.0005), 60 bias
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))
train: Scanning /content/drive/MyDrive/jai/yolov5/train/labels.cache... 0 images, 0 backgrounds, 113 corrupt: 100% 113/113 [00:00<?, ?it/s]
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/14463439-group-of-young-people-on-a-white-background.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/14463439-group-of-young-people-on-a-white-background.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/15618058695_65f33b01a0_b.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/15618058695_65f33b01a0_b.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/1573374342219.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/1573374342219.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/166247399_e4f08b.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/166247399_e4f08b.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/1711052955.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/1711052955.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/20160614-adults.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/20160614-adults.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/214725336_230026935792934_2671056484030306635_n.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/214725336_230026935792934_2671056484030306635_n.jpg'

**Reply by glenn-jocher** [reply]
Hello,

It seems like you're encountering issues with corrupt images during your training process. Here are a few steps to help resolve this:

1. **Verify Image Integrity**: Before training, ensure all your images can be opened without errors. You can use image processing software or a script to check and remove any corrupt files.

2. **Check Image Paths**: Ensure the paths in your dataset are correctly specified and accessible from your training environment.

3. **File Formats**: Confirm that all images are in a format supported by YOLOv5 (e.g., JPG, PNG). Convert any images that are not in a supported format.

4. **Re-download Images**: If some images are frequently causing issues, try re-downloading them or replacing them with alternative sources.

5. **Reduce Image Size**: If your images are very large, consider resizing them to a manageable size before training.

For the AMP issue, it's disabled due to a check failure, which might be related to the corrupt images or specific configurations in your setup. Once the image issues are resolved, AMP might work as expected.

If the problem persists after trying these steps, please provide more details about your setup and any error logs for further assistance.

**Comment by akhileshkumar1209** [comment]
i am facing the issue in my custom model training i have tried more then 10 times to train the model but in 8 time it is showing corrupt images as error how can i fix it.
Transferred 342/349 items from yolov5s.pt
AMP: checks failed ❌, disabling Automatic Mixed Precision. See ultralytics/yolov5#7908
optimizer: SGD(lr=0.01) with parameter groups 57 weight(decay=0.0), 60 weight(decay=0.0005), 60 bias
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))
train: Scanning /content/drive/MyDrive/jai/yolov5/train/labels.cache... 0 images, 0 backgrounds, 113 corrupt: 100% 113/113 [00:00<?, ?it/s]
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/14463439-group-of-young-people-on-a-white-background.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/14463439-group-of-young-people-on-a-white-background.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/15618058695_65f33b01a0_b.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/15618058695_65f33b01a0_b.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/1573374342219.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/1573374342219.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/166247399_e4f08b.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/166247399_e4f08b.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/1711052955.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/1711052955.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/20160614-adults.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/20160614-adults.jpg'
train: WARNING ⚠️ /content/drive/MyDrive/jai/yolov5/train/images/214725336_230026935792934_2671056484030306635_n.jpg: ignoring corrupt image/label: cannot identify image file '/content/drive/MyDrive/jai/yolov5/train/images/214725336_230026935792934_2671056484030306635_n.jpg'

**Reply by glenn-jocher** [reply]
It looks like you're encountering issues with corrupt images during your training process. Here are a few steps you can take to resolve this:

1. **Verify Image Integrity**: Before training, ensure all your images are in a proper format and not corrupted. You can use image processing tools or scripts to check and remove any corrupt files.

2. **Check Paths**: Ensure the paths in your dataset configuration file are correct and accessible. Sometimes, incorrect paths or permissions can cause similar errors.

3. **Update YOLOv5**: Make sure you are using the latest version of YOLOv5, as updates often include bug fixes and improvements that might resolve your issue.

4. **Reduce Dataset Size**: Try training with a smaller subset of your dataset to pinpoint if specific images or batches are causing the issue.

If the problem persists, consider providing more details or sharing your dataset configuration for further assistance.

**Comment by LGCCoding** [comment]
Dear developers, I would like to ask, I have my own training set, but every training results are different, whether using pre-training weights or not, I would like to ask if there is any way to ensure that the results can be repeated under the same configuration, thank you very much, this problem has troubled me for a long time.

**Reply by glenn-jocher** [reply]
Hello,

Thank you for reaching out with your query about achieving consistent training results with YOLOv8. To ensure reproducibility in your model training, you can follow these steps:

1. **Set a Fixed Seed**: Before starting your training, set a fixed seed for all random number generators in Python, NumPy, and PyTorch. This helps in generating the same random numbers across different training sessions, which is crucial for consistency.

   ```python
   import torch
   import numpy as np
   import random

   seed = 42
   torch.manual_seed(seed)
   np.random.seed(seed)
   random.seed(seed)
   if torch.cuda.is_available():
       torch.cuda.manual_seed_all(seed)
   ```

2. **Control Source of Non-determinism**: Some PyTorch operations are non-deterministic by nature, especially on GPUs. You can control this by setting:

   ```python
   torch.backends.cudnn.deterministic = True
   torch.backends.cudnn.benchmark = False
   ```

3. **Consistent Data Loading**: Ensure that the data loading process is consistent across runs. This involves using the same data splits and ensuring that any data augmentation processes are deterministic or controlled.

4. **Use Pretrained Weights**: Starting from the same pretrained weights for each training session can also help in achieving consistent results.

By following these steps, you should be able to see more consistent results across your training runs. If you continue to experience variability, it might be useful to check if any part of your data pipeline or model configuration changes between runs.

For more detailed guidance, please refer to our documentation on [training custom data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/).

**Comment by krausekeiii** [comment]
Hello all, can anyone point me in the direction of the documentation explaining the column titles in the results.csv file, specifically the one generated by YOLOv10, but any explanation will be greatly appreciated.

**Reply by pderrenger** [reply]
@krausekeiii hello!

Thank you for reaching out. The `results.csv` file generated by YOLOv10 (and other YOLO versions) contains various metrics and statistics that are logged during the training process. Here's a brief overview of the common columns you might find in this file:

- **epoch**: The current epoch number.
- **train/box_loss**: The bounding box regression loss during training.
- **train/obj_loss**: The objectness loss during training.
- **train/cls_loss**: The classification loss during training.
- **metrics/precision**: The precision metric on the validation set.
- **metrics/recall**: The recall metric on the validation set.
- **metrics/mAP_0.5**: The mean Average Precision (mAP) at IoU threshold 0.5.
- **metrics/mAP_0.5:0.95**: The mean Average Precision (mAP) averaged over IoU thresholds from 0.5 to 0.95.
- **val/box_loss**: The bounding box regression loss on the validation set.
- **val/obj_loss**: The objectness loss on the validation set.
- **val/cls_loss**: The classification loss on the validation set.

These columns provide a comprehensive view of the model's performance and training progress. For a more detailed explanation of each metric and how they are calculated, you can refer to the [Ultralytics documentation](https://docs.ultralytics.com).

If you have any specific questions about the columns or need further clarification, feel free to ask! 😊

**Comment by Syed-bisma2** [comment]
Dear developers, I would like to ask for help training a Yolov7 model for plant disease detection...as I am new to it I do not know how to start even from the data loading up to saving the model....how to show the confusion matrix , graphs, and all...I don't how to train a model... I know only up to annotation.

**Reply by glenn-jocher** [reply]
@Syed-bisma2 to get started with training a YOLOv7 model for plant disease detection, you'll need to prepare your dataset in the YOLO format, train the model using the Ultralytics library, and visualize results like confusion matrices. You can find a detailed guide on training custom data in the Ultralytics documentation. If you have specific questions, feel free to ask here.

**Comment by NikilXYZ** [comment]
I've got some custom splits (eg. I'm trying to quantize my model, so I've got splits named quantize, finetune, eval). I want to use yolov5 val.py for this. Can I have such custom split names in my dataset yaml? How can I make sure val.py uses the right split?

**Reply by NikilXYZ** [reply]
Seems like check_dataset is only looking for specific train, val, test, download

**Reply by glenn-jocher** [reply]
@NikilXYZ the `val.py` script in YOLOv5 expects specific keys like `train`, `val`, and `test` defined in your dataset YAML file. Custom split names like `quantize` or `finetune` are not natively supported. To use your custom splits, you can rename them to `val` or temporarily modify the `val.py` script to accommodate your custom split names. Make sure the paths in your dataset YAML correctly point to the respective datasets.

**Comment by AashishThakur1321** [comment]
my data is in .npy grayscale format i want to change the data loading pipeline to adjust to .npy file but i am not able to locate where to find the data loading part from the file paths

**Reply by glenn-jocher** [reply]
@AashishThakur1321 to modify the data loading pipeline for `.npy` grayscale files, you can start by reviewing and editing the `Dataset` class in the repository's `datasets.py` file. This is where the data loading logic is defined. Ensure you implement a custom loader to handle `.npy` files properly. For further guidance on dataset preparation, refer to [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/). Let me know if you need additional pointers!

**Comment by AashishThakur1321** [comment]
i have trained my model to detect objects but it is not able to detect small object, how can i improve my accuracy to detect small objects?

**Reply by glenn-jocher** [reply]
@AashishThakur1321 to improve the detection of small objects, you can try increasing the input image size during training (e.g., `--img 1280`), ensuring your dataset has sufficient examples of small objects, and tuning hyperparameters like `anchor_sizes` for better small object detection. Additionally, consider augmenting your dataset with smaller object instances. For further details, refer to [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/). Let me know if you have more questions!

**Comment by hoseinalhoseiny** [comment]
hi , how to write scripts related to load Datasets and Training Images and Validation Images and corresponding labels files?

**Reply by hoseinalhoseiny** [reply]
i am facing issue during training Yolov5 model  for detect walls in 22 images. how can i fix it?
Transferred 343/349 items from yolov5s.pt
optimizer: SGD(lr=0.01) with parameter groups 57 weight(decay=0.0), 60 weight(decay=0.0005), 60 bias
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
train: Scanning /content/gdrive/MyDrive/WallDetection_Dataset/train/labels... 18 images, 0 backgrounds, 13 corrupt: 100% 18/18 [00:02<00:00,  6.08it/s]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan1.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0469      1.3792      1.2748      1.1728      1.1991      1.3218      1.0729      1.1055       1.106       1.194      1.1388      1.1038      1.1058      1.3527       1.106      1.3793]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan10.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.2343       1.238      1.2384      1.1477      1.2382      1.0885]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan11.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.1776      1.1981      1.1963      1.1356      1.0731      1.0747      1.0133      1.0736      1.0332        1.04      1.0332      1.0334      1.0332      1.0011]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan12.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.1026        1.73      1.0446         1.1      1.1002      1.1616      1.0587      1.3885      1.3565      1.4083       1.594      2.2719      2.4576      2.4565      2.0142      2.2337       2.234      1.1868      2.2351      1.2349      1.7922      1.5932      1.2352      1.0013      1.0056      1.0995
      1.0056      1.1999      1.2349      1.0855      1.1489]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan13.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.1416      1.2102      1.2102      1.1492      1.0881]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan14.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.2738      1.1106      1.2401      1.4405      1.4767      1.4772      1.4375      1.1691      1.0076      1.3357      1.4795       1.333      1.0201      1.4788]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan15.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.3399      1.2433      1.4771      1.0701      1.2433      1.3399]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan16.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.3157      1.3144      1.1448      1.2491      1.3212      1.3982      1.2491]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan17.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.1163      1.0497      1.0438      1.0434      1.1958      1.1958]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan3.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.1009      1.1002      1.0199      1.0288      1.0283      1.2912      1.2912      1.0123]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan6.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [      1.114      1.1561       1.121      1.0221      1.0839]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan8.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.1086      1.0278]
train: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/train/images/plan9.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [      1.034      1.0068      1.1832       1.212      1.1663      1.0424      1.1092      1.0158      1.0064      1.0069      1.1989      1.2114      1.1445      1.1441      1.1335      1.1525      1.1609      1.2336      1.2333      1.1605      1.1523      1.1305      1.1153      1.0301      1.0308      1.0204
      1.1526      1.0784      1.0711      1.0932      1.0936      1.0202      1.1262      1.1483      1.1488      1.1477      1.1397      1.0505      1.0198      1.0009      1.1119      1.1141      1.0424      1.0425      1.0134      1.1483      1.1483      1.0949      1.0078      1.0921      1.1522      1.0237
      1.2012      1.2414      1.2016       1.075      1.1135      1.1551      1.0627      1.1078]
train: New cache created: /content/gdrive/MyDrive/WallDetection_Dataset/train/labels.cache
val: Scanning /content/gdrive/MyDrive/WallDetection_Dataset/val/labels... 4 images, 0 backgrounds, 4 corrupt: 100% 4/4 [00:05<00:00,  1.29s/it]
val: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/val/images/plan20.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [      1.174      1.1747      1.2528      1.0729      1.0423]
val: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/val/images/plan21.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.5662      1.5664      1.2244      1.4309      1.5694      1.3064      1.3939      1.2097      1.1221]
val: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/val/images/plan22.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.2482      1.0926       1.017      1.0926      1.1873      1.3923]
val: WARNING ⚠️ /content/gdrive/MyDrive/WallDetection_Dataset/val/images/plan23.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.4408      1.2208      1.5075      1.4386      1.2761      1.2753      1.0828      1.0872      1.1184      1.2764]
val: New cache created: /content/gdrive/MyDrive/WallDetection_Dataset/val/labels.cache
Traceback (most recent call last):
  File "/content/yolov5/train.py", line 986, in <module>
    main(opt)
  File "/content/yolov5/train.py", line 688, in main
    train(opt.hyp, opt, device, callbacks)
  File "/content/yolov5/train.py", line 309, in train
    val_loader = create_dataloader(
                 ^^^^^^^^^^^^^^^^^^
  File "/content/yolov5/utils/dataloaders.py", line 184, in create_dataloader
    dataset = LoadImagesAndLabels(
              ^^^^^^^^^^^^^^^^^^^^
  File "/content/yolov5/utils/dataloaders.py", line 615, in __init__
    labels, shapes, self.segments = zip(*cache.values())
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 3, got 0)

**Reply by glenn-jocher** [reply]
@hoseinalhoseiny your labels have coordinates exceeding 1.0, indicating non-normalized values. YOLOv5 requires bounding box coordinates in **normalized xywh format** (0-1). Ensure all values in your label files are divided by image width/height. Check our [Train Custom Data guide](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) for dataset formatting details. With only 22 images, you'll likely need more data for effective training.
